src/libpmemobj/ulog.c

```diff
@@ -488,6 +488,18 @@ ulog_entry_buf_create(struct ulog *ulog, size_t offset, uint64_t gen_num,
 
 	pmemops_drain(p_ops);
 
+	/*
+	 * Allow having uninitialized data in the buffer - this requires marking
+	 * data as defined so that comparing checksums is not reported as an
+	 * error by memcheck.
+	 */
+#if VG_MEMCHECK_ENABLED
```

Comment:
change to `VALGRIND_DO_MAKE_MEM_DEFINED` and that should allow you to remove both ifs.

Reply:
I asked Igor to do this. With the previous version, On_valgrind was checked twice.